userCode/cache.py

```diff
@@ -38,9 +43,14 @@ def set(self, url: str, content: bytes, _ttl: Optional[timedelta] = None):
         except Exception:
             get_dagster_logger().warning(f"Unable to cache: {url}")
 
-    def get_or_fetch(self, url: str, force_fetch: bool = False) -> Tuple[bytes, int]:
+    def get_or_fetch(
+        self,
+        url: str,
+        force_fetch: bool,
+    ) -> Tuple[bytes, int]:
         # If we are in prod we want to ignore using the cache and not store anything
-        if not RUNNING_AS_TEST_OR_DEV():
+        runningInProd = not RUNNING_AS_TEST_OR_DEV
+        if (self.skip_caching_in_prod and runningInProd) or RUNNING_AS_TEST_OR_DEV:
```

Comment:
This is really confusing. We are using a lot of double negative on three booleans:
- `skip_caching_in_prod`
- `force_fetch`
- `RUNNING_AS_TEST_OR_DEV`
This is too many IMO